docs/tutorial/new-tutorial.md

```diff
@@ -0,0 +1,384 @@
+(tutorial)=
+# SQLair tutorial
+
+Imagine you are writing a Go program that handles your customers and their
+orders. The business logic is contained in the Go program but the customer and
+order data is stored in a SQL database. In this tutorial, we'll see how easy
+SQLair makes it to map the database between the database and the Go program.
+Let's get started!
+
+
+```{note}
+This tutorial will take around 30 minutes to complete. If you want a quicker
+start see {ref}`prepare-and-run`.
+```
+
+## Setting up the environment
+
+To start with create a new folder called tutorial and initialise a new Go project.
+```bash
+mkdir tutorial 
+cd tutorial
+go mod init sqlair-tutorial
+```
+
+We first need to set up the backbone of our project, the database. The easiest
+way to create a SQL database in Go is to use [SQLite](https://www.sqlite.org/)
+with the [Go driver from mattn](https://github.com/mattn/go-sqlite3) though
+SQLair works with any database driver that can interface with
+[database/sql](https://pkg.go.dev/database/sql). With SQLite, we can create an in
+process, in memory database that can be populated within the program and that
+will disappear when the program finishes running.
+
+We also want to set the database up with our customer and order schema. We will
+have two tables. One for customers and the other for orders.
+
```

Comment:
This whole paragraph is too much talking at the user and it's also too much detail about something that is not at issue. I'd simplify ("You'll also need a SQL database to store your customer and order data. Let's create one:", then in-code comments to clarify the steps). If you still feel the need to highlight something else (e.g., that SQLair works with any database driver that can interface with... -- though I'd also wonder why we're not highlighting that bit in the homepage), put that in a note (which, if you want, you can make exciting by prefixing it with "Did you know? ...").

Reply:
I've compressed that first paragraph to just mention the database and driver we are using. I think this is pretty important backgroud info since it affects the syntax of the SQL that people would write. I've moved the comment about in memory/in process into the code.

The point about it working with any driver should be on the homepage tbh.